app/src/ui/dialog/dialog.tsx

```diff
@@ -202,9 +202,21 @@ export class Dialog extends React.Component<IDialogProps, IDialogState> {
   }
 
   public componentDidMount() {
+    if (!this.dialogElement) {
```

Comment:
While I doubt that it's possible for the dialogElement to be null at this point I'm still not a fan of the sentiment of bailing on the whole method if it is. Could we instead wrap an `if (this.dialogElement !== null)` around the two lines that use it?

Reply:
Sure, I can do that. The reason why I chose to bail on the entire method is that, if `dialogElement` is unavailable for whatever reason, the rest of the method doesn't make sense:

1. The old code (before this PR) would encounter a runtime error due to `undefined.showModal()`;
2. `this.setState({ isAppearing: true })` wouldn't make sense since nothing is "appearing";
3. `this.scheduleDismissGraceTimeout()` also wouldn't make sense.

To keep compatibility with old behavior, would a better approach be to manually raise an exception if `dialogElement` was unavailable?

Reply:
I don't feel strongly enough about this. Let's keep it as-is, it has no practical effect at all either way